### DISCUSSION TITLE
How often is bugfix 2.0.x pulled

### DISCUSSION THREAD

**Discussion by jjfs127** [Jyers/Marlin#85] [Q&A] [answered]
Great work as always... 
Just wanted to know what's the interval like or how often are you likely to merge Marlin's 2.0.x bugfix into your Ext_UI branch? I wanted to give the meatpack plugin a go and seems like recent activity fixes the serial unable to connect problem with meatpack/octoprint.
Thanks

**Comment by t3chguy** [comment]
Pulling bugfix-2.0.x yourself isn't too difficult but agreed that it would be good that it gets done soon for the meatpack fix

**Reply by jjfs127** [reply]
Using the Windows Desktop Github client, that UI perplexes me. It is simply cloning the Marlin repo and merging the bugfix branch into Jyers Ext_Ui branch? I'm just worried about handling conflicts and all...

**Reply by Schnabulation** [reply]
Once you setup the upstream repo it‘s just `git merge upstream/bugfix-2.0.x`

I merge Jyers repo into mine like this...

**Reply by jjfs127** [reply]
I just installed the command line version of git on Windows.
```
  2.0.x
* Ender_3_V2_Extensible_UI
  config-changes
```

So:
Set up an upstream repo:

```
git remote add upstream https://github.com/MarlinFirmware/Marlin
```

Update:
```
git fetch upstream
```

Merge:
```
git merge upstream/bugfix-2.0.x
```

And tons of conflicts:
```
...
Auto-merging Marlin/src/HAL/STM32/fastio.h
CONFLICT (content): Merge conflict in Marlin/src/HAL/STM32/fastio.
Auto-merging Marlin/src/HAL/STM32/HAL_SPI.cpp
CONFLICT (content): Merge conflict in Marlin/src/HAL/STM32/HAL_SPI
Auto-merging Marlin/src/HAL/STM32/HAL.h
CONFLICT (content): Merge conflict in Marlin/src/HAL/STM32/HAL.h
Auto-merging Marlin/src/HAL/STM32/HAL.cpp
CONFLICT (content): Merge conflict in Marlin/src/HAL/STM32/HAL.cpp
Auto-merging Marlin/src/HAL/SAMD51/QSPIFlash.cpp
Auto-merging Marlin/src/HAL/SAMD51/HAL.h
Auto-merging Marlin/src/HAL/LPC1768/inc/SanityCheck.h
Auto-merging Marlin/src/HAL/LPC1768/HAL.h
CONFLICT (content): Merge conflict in Marlin/src/HAL/LPC1768/HAL.h
Auto-merging Marlin/src/HAL/LINUX/HAL.h
Auto-merging Marlin/src/HAL/HAL.h
Auto-merging Marlin/src/HAL/ESP32/HAL.h
Auto-merging Marlin/src/HAL/DUE/timers.cpp
Auto-merging Marlin/src/HAL/DUE/eeprom_flash.cpp
Auto-merging Marlin/src/HAL/DUE/HAL_SPI.cpp
Auto-merging Marlin/src/HAL/DUE/HAL.h
Auto-merging Marlin/src/HAL/AVR/HAL.h
CONFLICT (content): Merge conflict in Marlin/src/HAL/AVR/HAL.h
Auto-merging Marlin/Version.h
Auto-merging Marlin/Configuration_adv.h
CONFLICT (content): Merge conflict in Marlin/Configuration_adv.h
Auto-merging Marlin/Configuration.h
CONFLICT (content): Merge conflict in Marlin/Configuration.h
Automatic merge failed; fix conflicts and then commit the result.
```

@t3chguy Thoughts?

Perhaps @Schnabulation you meant merge Jyers into bugfix? If so, less conflcts, just accept incoming?
```
UU Marlin/src/lcd/buttons.h
UU Marlin/src/lcd/dwin/e3v2/dwin.cpp
UU Marlin/src/lcd/dwin/e3v2/dwin.h
```

**Reply by Schnabulation** [reply]
Actually I have not yet merged bugfix-2.0.x into Jyers or the other way around - I just wanted to help you with the command line inputs. I personally have created my own fork from Jyers ExtUI branch and added my own config files. I merge his branch into mine whenever he makes changes - so I don‘t need to change the config files all the time.

However I have not yet tried to merge Jyers and official bugfix-2.0.x - I leave that to Jyers himself and trust him that he does it frequently. If you want to try I would suggest pulling bugfix and merging Jyers into that...

**Reply by jjfs127** [reply]
@Schnabulation 
Thanks for the suggestion about creating a fork and merging Jyers, that makes a lot of sense.
About the bugfix, I did exactly as you suggested the second attempt and merged Jyers into bugfix and saw three updated files. 

```
UU Marlin/src/lcd/buttons.h
UU Marlin/src/lcd/dwin/e3v2/dwin.cpp
UU Marlin/src/lcd/dwin/e3v2/dwin.h
```

I pretty much accepted incoming changes and it built correctly, just too chicken to try it out...

**Comment by Jyers** [comment, marked as answer]
To be honest, I've only been putting it off because I'm still pretty new to GitHub and was worried I wouldn't do it right. I just merged, so hopefully nothing got messed up :P

**Comment by DragRedSim** [comment]
Is there any plans to pull in the latest content from bugfix-2.0.* again? There appear to have been some changes on their end to the handling of the DWIN code, along with some other important fixes.

**Reply by Jyers** [reply]
The DWIN code from this branch is not only completely different from the main branch, but is also in an entirely different location as well. So DWIN changes in the main branch will not affect this branch. I try to pull from bugfix when there are relevant fixes for the E3V2. I'll take a look at the latest fixes to see if a pull would be a good idea.

**Reply by t3chguy** [reply]
@Jyers there are Meatpack improvements as of 2 weeks ago, would be good to get those in

**Reply by Jyers** [reply]
Oh nice! Yeah that would be good.

**Reply by tommymonk** [reply]
On this topic I was wondering if commits from this bug were merged, it might be the answer to [#169](https://github.com/Jyers/Marlin/issues/169)  for those of us who have tried everything.
https://github.com/MarlinFirmware/Marlin/pull/21283

**Reply by Jyers** [reply]
This has definitely been merged into this branch as the last merge was around a week ago I believe. What exact issues are you still facing?

**Reply by tommymonk** [reply]
When printing through Octoprint I get random print failures with serial exceptions, same `usb_serial_generic_read_bulk_callback - urb stopped: -32` error. Tried lots of things;

- New official power supply for the pi4
- New shielded usb cable
- Chokes on all power and data cables
- USB 5v masking
- Compiled my own binary
- Don't use Meatpack

**Reply by Jyers** [reply]
Interesting, unfortunately I don't have a ton of incite into what is causing this for you. All I can say is that I had this issue and fixed it by shielding my own cable. I know it sounds bad, but I literally covered it in some aluminum foil and wrapped it in electrical tape. I grounded the foil to the connector and haven't seen issues since. Given what you've said, It sounds like you've ruled out interference, but I figured I'd share my solution in case it could help. Also as I'm sure you're aware, don't run your usb anywhere near your heater wires/power supply cable.

**Reply by tommymonk** [reply]
Thanks for the tip, I'll try your tinfoil recipe and see if it works, desperate times call for desperate measures 😉
